engine/src/flutter/impeller/display_list/canvas.cc

```diff
@@ -1681,6 +1684,25 @@ std::shared_ptr<Texture> Canvas::FlipBackdrop(Point global_pass_position,
   return input_texture;
 }
 
+std::shared_ptr<Texture> Canvas::GetOrUploadTexture(
+    const sk_sp<const flutter::DlImage>& image) {
+  if (image->isDeferredUpload()) {
+    std::shared_ptr<CommandBuffer> cmd_buffer =
+        renderer_.GetContext()->CreateCommandBuffer();
+    std::shared_ptr<BlitPass> blit_pass = cmd_buffer->CreateBlitPass();
+    std::shared_ptr<DeviceBuffer> device_buffer = image->GetDeviceBuffer();
+    blit_pass->AddCopy(DeviceBuffer::AsBufferView(device_buffer),
+                       image->impeller_texture());
+    image->SetUploaded();
```

Comment:
Shouldn't this be after the EnqueueCommandBuffer?

Reply:
Done